pkg/ring/lifecycler.go

```diff
@@ -56,12 +56,12 @@ type LifecyclerConfig struct {
 	FinalSleep       time.Duration `yaml:"final_sleep"`
 	TokensFilePath   string        `yaml:"tokens_file_path"`
 	Zone             string        `yaml:"availability_zone"`
+	SkipUnregister   bool          `yaml:"skip_unregister"`
```

Comment:
I'd suggest calling it UnregisterFromRing, and make it true by default (for compatibility).